### PR TITLE
Add A.S. funded food field to event form

### PR DESCRIPTION
**Pull request by SheepTester**
# Info

Depends on acmucsd/membership-portal#456

This allows the marketing team to populate the food field for events. `foodItems` isn't used anywhere else in the UI yet, but letting it be populated helps to avoid blocking generating AS attendance QR codes to the Discord bot.

## Changes

- Update types in apiRequests and apiResponses
- Add a field to the create and edit events form

# Type of Change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) 

# Testing

I have tested that my changes fully resolve the linked issue ...

- [ ] locally on Desktop.
- [x] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have added new Cypress tests that are passing.

# Checklist

- [x] I have performed a self-review of my own code.
- [x] I have followed the style guidelines of this project.
- [x] I have documented any new functions in `/src/lib/*` and commented hard to understand areas
      anywhere else.
- [x] My changes produce no new warnings.

# Screenshots

{DC84B23C-7039-4ABC-9F3D-D02B8FE85109}

### CODE REVIEW

File: src/lib/types/apiResponses.ts [hunk 1]
```diff
@@ -133,6 +133,7 @@ export interface PublicEvent {
   requiresStaff: boolean;
   staffPointBonus: number;
   discordEvent: string | null;
+  foodItems: string;
```
Comment (newracket) [inline src/lib/types/apiResponses.ts]: The foodItems field is nullable, so I think this should be `string | null`?

**Review by newracket — CHANGES_REQUESTED** [review]
Looks pretty good to me, just one small change

File: src/lib/types/apiResponses.ts (hunk 1 above)
Comment (SheepTester) [inline src/lib/types/apiResponses.ts]:
```suggestion
  foodItems: string | null;
```

File: src/lib/types/apiRequests.ts [hunk 2]
```diff
@@ -165,6 +165,7 @@ export interface OptionalEventProperties {
   requiresStaff?: boolean;
   staffPointBonus?: number;
   discordEvent: string | null;
+  foodItems?: string;
```
Comment (SheepTester) [inline src/lib/types/apiRequests.ts]:
```suggestion
  foodItems?: string | null;
```

**Review by newracket — APPROVED** [review]
LGTM

**Review by alexzhang1618 — APPROVED** [review]

**Pull request merged**